app/src/main/java/com/example/anfcodetest/MainPresenter.kt

```diff
@@ -1,103 +1,51 @@
 package com.example.anfcodetest
 
-import android.content.Intent
-import android.text.Html
-import android.text.method.LinkMovementMethod
-import android.view.View.GONE
-import com.squareup.picasso.Picasso
-import kotlinx.android.synthetic.main.card_row.view.*
-
-
-class MainPresenter : MainContract.Presenter {
-
-    override fun cardSetup(holder: CustomViewHolder, position: Int, cardList: List<Card>) {
-        val card = cardList[position]
```

Comment:
If we move this method to the ViewHolder itself, we will no longer have to iterate over our card list, or pass the whole thing through in the method!